node_modules/oae-core/resetpassword/js/resetpassword.js

```diff
@@ -16,144 +16,143 @@
-                            response = available;
-                        });
-                        return response;
+
+	// The widget container
+	var $rootel = $('#' + uid);
+	
+	/**
+	 * Verify whether or not the entered username already exists as a login id on the current tenant.
+	 * The ajax request in this function executes synchronously to allow a username to be checked for
+	 * existence before the user is created.
+	 *
+	 * @param  {String}     username                The username we want to check the existence of
+	 * @param  {Function}   callback                Standard callback function.
+	 * @param  {Boolean}    callback.available      Whether or not the username is available
+	 */
+	var isUserNameAvailable = function(username, callback) {
+            $.ajax({
+		url: '/api/auth/exists/' + username,
+		async: false,
+		success: function() {
+                    // The username already exists
+                    callback(true);
+		},
+		error: function(xhr, textStatus, thrownError) {
+                    // The username is still available
+                    callback(false);
+		}
+            });
+	};
+	
+	/**
+	 * Set up the validation on the register form, including the error messages
+	 */
+	var setUpValidation = function() {
+            var validateOpts = {
+		'rules': {
+                    'username': {
+			'minlength': 3,
+			'nospaces': true,
+			'validchars': true,
+			'usernameavailable': true
+                    }
+		},
+		'messages': {
+                    'username': {
+			'required': oae.api.i18n.translate('__MSG__PLEASE_ENTER_YOUR_USERNAME__'),
+			'minlength': oae.api.i18n.translate('__MSG__THE_USERNAME_SHOULD_BE_AT_LEAST_THREE_CHARACTERS_LONG__', 'register'),
+			'nospaces': oae.api.i18n.translate('__MSG__THE_USERNAME_SHOULDNT_CONTAIN_SPACES__')
+                    }
+		},
+		'methods': {
+                    'validchars': {
+			'method': function(value, element) {
+                            return this.optional(element) || !(/[<>\\\/{}\[\]!@#\$%\^&\*,:]+/i.test(value));
+			},
+			'text': oae.api.i18n.translate('__MSG__ACCOUNT_INVALIDCHAR__')
                     },
-                    //'text': oae.api.i18n.translate('__MSG__THIS_USERNAME_HAS_ALREADY_BEEN_TAKEN__')
-                    'text': "Username does not exist."
-                }
-            },
-            'submitHandler': getSecret
-        };
-        oae.api.util.validation().validate($('#resetpassword-input-username', $rootel), validateOpts);
-    };
+                    'usernameavailable': {
+			'method': function(value, element) {
+                            var response = false;
+                            isUserNameAvailable(value, function(available) {
+				response = available;
+                            });
+                            return response;
+			},
+			//'text': oae.api.i18n.translate('__MSG__THIS_USERNAME_HAS_ALREADY_BEEN_TAKEN__')
+			'text': "Username does not exist."
```

Comment:
This needs to be added to default.properties so translations can be added later.